internal/decryption/kmip/client.go

```diff
@@ -0,0 +1,385 @@
+// Copyright 2022 MongoDB Inc
```

Comment:
not blocking but this seems a lot of code that is not tested, would you be able to send some unit tests (perhaps on a follow-up pr)?

Reply:
doing this as part of https://jira.mongodb.org/browse/CLOUDP-116175 E2E smoke tests - KMIP